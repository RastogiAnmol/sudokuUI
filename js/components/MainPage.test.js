import React from "react";
import MainPageComponent from "./MainPage";
import Adapter from "enzyme-adapter-react-14";
import { shallow, mount, render, configure } from "enzyme";
configure({ adapter: new Adapter() });
describe("MainPageComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MainPageComponent debug />);

    expect(component).toMatchSnapshot();
  });
});
