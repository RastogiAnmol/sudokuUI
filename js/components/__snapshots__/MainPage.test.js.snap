// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Shall we play Sudoku
        </h2>,
        <Link
          className="buttonShowSol"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="new-game"
        >
          Let's start a new game
        </Link>,
      ],
      "className": "index",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Shall we play Sudoku",
        },
        "ref": null,
        "rendered": "Shall we play Sudoku",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Let's start a new game",
          "className": "buttonShowSol",
          "onlyActiveOnIndex": false,
          "style": Object {},
          "to": "new-game",
        },
        "ref": null,
        "rendered": "Let's start a new game",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Shall we play Sudoku
          </h2>,
          <Link
            className="buttonShowSol"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="new-game"
          >
            Let's start a new game
          </Link>,
        ],
        "className": "index",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Shall we play Sudoku",
          },
          "ref": null,
          "rendered": "Shall we play Sudoku",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Let's start a new game",
            "className": "buttonShowSol",
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "new-game",
          },
          "ref": null,
          "rendered": "Let's start a new game",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
